controller/api/destination/endpoint_translator.go

```diff
@@ -480,6 +490,74 @@ func toAddr(address watcher.Address) (*net.TcpAddress, error) {
 	}, nil
 }
 
+func createWeightedAddrForExternalWorkload(
+	address watcher.Address,
+	opaquePorts map[uint32]struct{},
+) (*pb.WeightedAddr, error) {
+	tcpAddr, err := toAddr(address)
+	if err != nil {
+		return nil, err
+	}
+
+	weightedAddr := pb.WeightedAddr{
+		Addr:         tcpAddr,
+		Weight:       defaultWeight,
+		MetricLabels: map[string]string{},
+	}
+
+	weightedAddr.MetricLabels = pkgK8s.GetExternalWorkloadLabels(address.OwnerKind, address.OwnerName, address.ExternalWorkload)
+	// If the address is not backed by an ExternalWorkload, there is no additional metadata
+	// to add.
+	if address.ExternalWorkload == nil {
+		return &weightedAddr, nil
+	}
```

Comment:
How likely is it to hit this given that we only call this function when the address does have an `ExternalWorkload` reference? Also, an external workload reference should be paramount to sending traffic to  a VM. Do you think we should return an error here instead of returning the address?